salt/modules/napalm_network.py

```diff
@@ -1914,7 +1914,7 @@ def load_template(template_name=None,
     salt_render_prefixes = ('salt://', 'http://', 'https://', 'ftp://')
     salt_render = False
     file_exists = False
-    if not isinstance(template_name, (tuple, list)):
+    if template_name and not isinstance(template_name, (tuple, list)):
```

Comment:
To make sure I understand, what we're looking for here is that we have something *in* the tuple or list? Because None will already fail the check, but empty lists will pass.

Wouldn't it make more sense to make this positively check that's text? I'm pretty sure we could use something like `six.text_type`, I forget what the name is exactly.

Reply:
None type will not fail the type check and cause the if block to be eval'd, which is the problem since at the function signature `template_name` defaults to None:
```
>>> template_name = None
>>> if not isinstance(template_name, (tuple, list)):
...   print(True)
...
True
```
I'm happy to explicitly check for text if you would like. The way this diff is written, anything other than a None, tuple or list type will cause the if block to be eval'd.